### PR TITLE
libfaketime: Modernize Portfile, remove patches

### PR DESCRIPTION
**Pull request by TheRealKeto**
#### Description

Modernize the Portfile by migrating to the Makefile PG, and remove unused patches, since their requirements for removal have been met.

###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.2 14C18

macOS 10.7.5 11G63 x86_64
Xcode 4.3.3 4E3002

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/libfaketime/Portfile
Comment (reneeotten) [inline sysutils/libfaketime/Portfile:10]: this doesn't change anything consequential do there is no need to increase the "revision". If anything, I would make changes like this when updating the version...

Comment (TheRealKeto) [reply sysutils/libfaketime/Portfile:10]: I'd forgotten about removing the patch that the port previously used. Does that count for a rev-bump since the build would be different?

Comment (reneeotten) [reply sysutils/libfaketime/Portfile:10]: Presumably the implementation in legacy-support would do the same thing? I didn't check, but let's see what @catap thinks about this change.

Comment (catap) [reply sysutils/libfaketime/Portfile:10]: @reneeotten yes, the new version of legacy support includes the fix, seems that no need for a patch.

@TheRealKeto have you tried it on some old system? Does it work?

Anyway, I'd like to keep revbump to force rebuild and relink against new legacy-support.

Comment (TheRealKeto) [reply sysutils/libfaketime/Portfile:10]: @catap Yes. I tried building the project on OSX Lion 10.7.5. More information is on the description of the PR. I can't vouch for other legacy systems, but given that it works on as low as Lion, it's good to go :)

Comment (catap) [reply sysutils/libfaketime/Portfile:10]: @TheRealKeto just build or also use it? :)

**Review by catap — APPROVED** [review]
With assumption that @TheRealKeto had run `port test` on 10.7, let merge it

**Comment by TheRealKeto** [comment]
Tests failed for me on OSX 10.7.5, but the `faketime` binary worked when being installed.

**Comment by TheRealKeto** [comment]
Tests failed for me on OSX 10.7.5, but the `faketime` binary worked when being installed. [Here's a log of the results](https://github.com/macports/macports-ports/files/13303656/main.log).

**Comment by TheRealKeto** [comment]
Tests failed for me on OSX 10.7.5, but the `faketime` binary worked when being installed. [Here's a log of the results](https://github.com/macports/macports-ports/files/13303656/main.log). It's important to know that running `port test` returns the same error featured on the linked log before this change.

File: sysutils/libfaketime/Portfile
Comment (TheRealKeto) [reply sysutils/libfaketime/Portfile:10]: @catap I tested the `faketime` binary after building and installing with a few dates (e.g `faketime -f "1999-01-01 00:00:00"`), as well as whether it could still "hook" into system binaries (given that I was also on OSX Lion)

**Comment by catap** [comment]
@TheRealKeto thanks and fine for me. Unfortunately, I haven't got the commiter right (and I've asked 3 times with interval a year), so, let's wait until someone with the right will merge it.

@reneeotten, @herbygillot may I ask you to do it?

**Pull request merged**
